### ISSUE TITLE
youtube-dl not working on Gentoo

### ISSUE THREAD

**Issue by SimplyCorbett** [Chocobozzz/PeerTube#4430]
Recently swapped to Gentoo because I like to tinker. Transferred the installation successfully and everything works other then youtube-dl.

I'm assuming this is because like on debian 10 you need python-is-python3 or python-is-python2. These packages don't exist in Gentoo.

Can you update the install instructions to include the required packages under the Gentoo area in any os? Thank you!

**Comment by rigelk** [comment]
Hi @SimplyCorbett,

None of us works with Gentoo ; this is mainly a part maintained by the community, so unless someone steps up to correct it, it will stay as is (or be deleted).

Please also use the documentation template.

**Comment by SimplyCorbett** [comment]
Hi @rigelk 

I understand. However I feel that this is a bug that should be fixed regardless. youtube-dl should work out of the box without a workaround solution like python-is-python2.

As far as gentoo goes peertube is fully functional on it minus the above issue, so removing all support from the manual/documentation seems premature.

**Comment by rigelk** [comment]
You're welcome to PR a doc fix at https://framagit.org/framasoft/peertube/documentation

> youtube-dl should work out of the box without a workaround solution like python-is-python2.

Isn't it a question better fit for the youtube-dl community?

**Comment by frankstrater** [comment]
Hi @SimplyCorbett 

I'll update the dependencies guide for Gentoo soon. For all the distros that don't have the `python-is-python2` or `python-is-python3` package in their respective repositories, you can 'fix' this by doing:

`sudo ln -s /usr/bin/python2 /usr/bin/python`

or

`sudo ln -s /usr/bin/python3 /usr/bin/python`

depending on the Python version you have installed.

**Comment by frankstrater** [comment]
@rigelk

Which part of the dependencies guide does the PeerTube team maintain? Just the Debian/Ubuntu one?

Maybe it's a good idea that it's mentioned that the dependencies guide is maintained by the community (some note in the intro of that page). I've seen a lot of frustration in the community about the sometimes outdated dependencies guide and you are constantly blamed for it.

**Comment by rigelk** [comment]
> Which part of the dependencies guide does the PeerTube team maintain? Just the Debian/Ubuntu one?

Pretty much, yes.

> Maybe it's a good idea that it's mentioned that the dependencies guide is maintained by the community

Maybe. Invoking @SansPseudoFix there.

**Comment by frankstrater** [comment]
@SimplyCorbett Could you confirm my suggested 'fix' works on Gentoo? I don't have a Gentoo install myself and I assume the current default slot for Python (dev-lang/python) is Python3.

**Comment by SimplyCorbett** [comment]
@frankstrater Yes it did fix it.

**Comment by SansPseudoFix** [comment]
Is https://github.com/Chocobozzz/PeerTube/pull/4440 ok?

**Comment by kleph** [comment]
Hi from a fellow gentoo user, 

I am puzzled with that solution.
`/usr/bin/python` symlinks should not be handled manually on gentoo.

This is probably no too dangerous though, I think the worst it could do is confuse a bit gentoo's internal tool.
But since python's installations and different versions are handled by the distribution itself, those links are updated each time the default version of python is modified (either when a new version is rolled in or when the user change it with `python-exec.conf` or formerly `eselect python`) and should already exists.

Here is the Gentoo's documentation :  https://wiki.gentoo.org/wiki/Python#Configuration

Also, `dev-lang/python` is probably useless as dependency, it's very unlinkely that you can have a working gentoo without python ;-)  (emerge, and other gentoo internal tools are written in python)

@SimplyCorbett   What was your original problem with youtube-dl ?
It's a package provided and maintained within gentoo, `net-misc/youtube-dl`. I think it would be better to just add it as a dependency in the set.

**Comment by frankstrater** [comment]
The included youtube-dl binary in Python is written in Python and has this as interpreter directive:

`#!/usr/bin/python`

@kleph Any suggestions on a better way to make this work on Gentoo when you can't change this directive and this file location doesn't exist?

**Comment by kleph** [comment]
Yes, `python-exec` is the way to change the default python interpreter, AKA where the link `/usr/bin/python` points to.

Insted of calling `ln -s` manually,  `emerge -1 python-exec` will do the same job, but "the gentoo way".
It will link the version defined in the file `/etc/python-exec/python-exec.conf` to `/usr/bin/python`, so I think it should be safe and will regenerate the symlink if it's missing or if it's pointing to an old uninstalled version of python.
